### PR TITLE
Contamination filtering

### PR DESCRIPTION
**Pull request by CKComputomics**
Adds an optional contamination filtering step.
Enables the user to filter contaminants from the fastq files by supplying different fast files that contain contaminant sequences. This is enabled using `--filter_contamination`. The individual pasta files can be supplied using `--rrna`,   `--trna`, `--cdna`, `--ncrna`, `--pirna`. A additional, unclassified, database can be used with `--other_contamination`. The fasta files for cDNA, ncRNA, piRNA, and others are filtered using the miRNA database to remove possible overlap.
## PR checklist

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](<https://github.com/>nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint .`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [X] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by apeltzer** [comment]
@CKComputomics  - we need to fix the editorconfig lint checks. Prettier is fine, but you're having multiple new lines in the files mentioned in the CI that need to be addressed 👍🏻

**Comment by CKComputomics** [comment]
> @CKComputomics - we need to fix the editorconfig lint checks. Prettier is fine, but you're having multiple new lines in the files mentioned in the CI that need to be addressed 👍🏻

**Review by apeltzer — APPROVED** [review]
Minor points, please just merge the suggestions - otherwise looks all fine 👍🏻

File: docs/output.md [hunk 1]
```diff
@@ -58,6 +59,12 @@ This is an example of the output we can get:
 
 ![cutadapt](images/cutadapt_plot.png)
 
+## Bowtie2
+
+[Bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml) is used to align the reads to user-defined databasesof contaminants.
```
Comment (apeltzer) [inline docs/output.md:64]:
```suggestion
[Bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml) is used to align the reads to user-defined databases of contaminants.
```

**Pull request closed without merging**